packages/twenty-front/src/modules/object-record/record-index/options/hooks/useExportTableData.ts

```diff
@@ -144,25 +152,33 @@ export const useExportTableData = ({
   const { totalCount, records, fetchMoreRecords } = useFindManyRecords({
     ...usedFindManyParams,
     limit: pageSize,
-    onCompleted: (_data, options) => {
-      setHasNextPage(options?.pageInfo?.hasNextPage ?? false);
-    },
   });
 
   useEffect(() => {
-    const MAXIMUM_REQUESTS = Math.min(maximumRequests, totalCount / pageSize);
+    const MAXIMUM_REQUESTS = isDefined(totalCount)
+      ? Math.min(maximumRequests, totalCount / pageSize)
+      : maximumRequests;
 
     const downloadCsv = (rows: object[]) => {
       csvDownloader(filename, { rows, columns });
       setIsDownloading(false);
-      setProgress(undefined);
+      setProgress({
+        count: undefined,
+        type: 'number',
+      });
     };
 
     const fetchNextPage = async () => {
       setInflight(true);
+      setPreviousRecordCount(records.length);
       await fetchMoreRecords();
       setPageCount((state) => state + 1);
-      setProgress(percentage(pageCount, MAXIMUM_REQUESTS));
+      setProgress({
+        count: totalCount
+          ? percentage(pageCount, MAXIMUM_REQUESTS)
```

Comment:
I would rather always put the exportedRecordCount in the count and display the percentage at display. Let's also store totalRecordCount in export progress if needed